framework/decode/dx12_object_info.h

```diff
@@ -137,12 +140,33 @@ static size_t GetResourceValueSize(ResourceValueType type)
     }
 }
 
+struct ArgumentBufferExtraInfo
+{
+    D3D12CommandSignatureInfo* command_signature_info{ nullptr };
+    DxObjectInfo*              argument_buffer{ nullptr };
+    uint64_t                   argument_buffer_offset{ 0 };
+};
+
 struct ResourceValueInfo
 {
     uint64_t              offset{ 0 };
     ResourceValueType     type{ ResourceValueType::kUnknown };
     uint64_t              size{ 0 };
     D3D12StateObjectInfo* state_object{ nullptr }; ///< Used to map values in shader records.
+    ArgumentBufferExtraInfo arg_buffer_extra_info;
+
+    ResourceValueInfo(uint64_t                in_offset,
+                      ResourceValueType       in_type,
+                      uint64_t                in_size,
+                      D3D12StateObjectInfo*   in_state_object          = nullptr,
+                      ArgumentBufferExtraInfo in_arg_buffer_extra_info = { nullptr, nullptr, 0 })
```

Comment:
These should not have default values. The default arguments to the constructor can lead to errors where new `ResourceValueInfo`s are created and modified from existing `ResourceValueInfo`s.

For example in `CopyResourceValuesFromDstToSrc` below:
```
        src.insert({ static_cast<uint64_t>(src_offset),
                     (*dst_iter).type,
                     static_cast<uint64_t>((*dst_iter).size),
                     (*dst_iter).state_object });
```
`ResourceValueInfo::in_arg_buffer_extra_info` from `(*dst_iter)` should be set on the new `ResourceValueInfo` but currently it gets the default argument instead.